lib/Bindings/Python/support.py

```diff
@@ -143,6 +143,10 @@ def attribute_to_var(attr):
     return ir.IntegerAttr(attr).value
   except ValueError:
     pass
+  try:
+    return ir.StringAttr(hw.InnerSymAttr(attr).symName).value
```

Comment:
We'll soon want to support more than just the default symName, and probably have this some opaque attribute instead of explicitly a string (even before we make these not strings internally-- to reduce churn on the python side).

I expect this helps get things working in the short term, but concerned about adding more code (+users) that will need to be updated again "soon".  Don't mean to block at all, just want to mention it before this gets used and relied on in code that we know we want to change.

Although it'd be nice if the thing called `InnerSym*` was one inner symbol, not a collection of them, or I guess I'm saying sorting out to make this good for python is not unrelated to improving how this works/looks in C++-land too :smile: .  Haven't really looked at how/what python code is doing with these, I'll take a look...

Reply:
On the python side, if we get an `InnerRefAttr`, we need to be able to check whether it points to an object that has a symbol. For now it is fine to keep this as a string as we can remove the hack from `circt-python`, but we should devise an opaque type that supports this operation. We should not leak out string names.

Reply:
Yeah, the concern totally makes sense. The PR fixes the issue @nandor encountered in the internal code that is currently workaround by callsites of `attribute_to_var`. Currently CAPI/Python binding only supports [single string representation] so I believe multiple symbols won't work with the current implementation. 

I agree that ideally it's better not to introduce new stuffs, but I think there has been already several users of InnerSymAttr across python codebase so we'll need to replace them anyway.